cmd/apply.go

```diff
@@ -0,0 +1,223 @@
+package cmd
+
+import (
+	"bytes"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	"github.com/kubernetes-sigs/ingress-controller-conformance/internal/pkg/assets"
+	"github.com/kubernetes-sigs/ingress-controller-conformance/internal/pkg/k8s"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/printers"
+	"k8s.io/kubectl/pkg/cmd/apply"
+	"k8s.io/kubectl/pkg/cmd/util"
+)
+
+var applyIngressClass string
+
+func newStandardIO() genericclioptions.IOStreams {
+	return genericclioptions.IOStreams{
+		In:     os.Stdin,
+		Out:    os.Stdout,
+		ErrOut: os.Stderr,
+	}
+}
+
+// complete verifies if ApplyOptions are valid and without conflicts by cribbing from (*apply.ApplyOptions)Complete().
+func complete(o *apply.ApplyOptions, f util.Factory) error {
+	var err error
+
+	o.FieldManager = "ingress-controller-conformance"
+
+	o.DryRun = false
+	o.ForceConflicts = false
+	o.ServerDryRun = false
+	o.ServerSideApply = false
+
+	// Don't bother recording the apply in an attribute.
+	o.Recorder = &genericclioptions.NoopRecorder{}
+
+	// allow for a success message operation to be specified at print time
+	o.ToPrinter = func(operation string) (printers.ResourcePrinter, error) {
+		o.PrintFlags.NamePrintFlags.Operation = operation
+		if o.DryRun {
+			o.PrintFlags.Complete("%s (dry run)")
+		}
+		if o.ServerDryRun {
+			o.PrintFlags.Complete("%s (server dry run)")
+		}
+		return o.PrintFlags.ToPrinter()
+	}
+
+	o.DiscoveryClient, err = f.ToDiscoveryClient()
+	if err != nil {
+		return err
+	}
+
+	dynamicClient, err := f.DynamicClient()
+	if err != nil {
+		return err
+	}
+
+	// NOTE(jpeach): Here we omit calling (*FilenameOptions)
+	// RequireFilenameOrKustomize() which enforces that either the
+	// '-f' or '-k' flags must be present. We manually specified
+	// the resources to parse from the compiled assets, so we don't
+	// need either.
+
+	o.DeleteOptions = o.DeleteFlags.ToOptions(dynamicClient, o.IOStreams)
+
+	o.OpenAPISchema, _ = f.OpenAPISchema()
+	o.Validator, err = f.Validator(true)
+	if err != nil {
+		return err
+	}
+
+	o.Builder = f.NewBuilder()
+	o.Mapper, err = f.ToRESTMapper()
+	if err != nil {
+		return err
+	}
+
+	o.DynamicClient, err = f.DynamicClient()
+	if err != nil {
+		return err
+	}
+
+	o.Namespace, o.EnforceNamespace, err = f.ToRawKubeConfigLoader().Namespace()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// addObjectLabels adds the given map of key-value pairs to
+// the object as labels. Any existing labels for these keys
+// will be overwritten.
+func addObjectLabels(obj v1.Object, kv map[string]string) {
+	labels := obj.GetLabels()
+	if labels == nil {
```

Comment:
[aside] I think this actually doesn't ever return nil, but better safe than sorry.